### DISCUSSION TITLE
divers dev

### DISCUSSION THREAD

**Discussion by FranckFreiburger** [FranckFreiburger/vue3-sfc-loader#29] [French]
@Toilal, a propos du pb. "vue 3 > custom template language", voici ce qui est attendu pour vue3 :

```
    options.moduleCache.custom = { render: (source, options, callback)  => {

      callback(null, source + '_modified')
    }}
```

Je pense que tu va devoir rendre ça asynchrone :

https://github.com/FranckFreiburger/vue3-sfc-loader/blob/ae53bb9191bacae8a09a7f197050228ced14d791/src/createVue2SFCModule.ts#L115

**Comment by Toilal** [comment]
Ok, bien compris. L'API est quand même complexe à cause de tout ça :(

Sinon j'essaye de lancer les tests sur la dernière version, mais j'ai cette erreur

    pageerror [Error: TypeError: cloneDeep is not a function
        at normalizeFile (http://local/vue3-sfc-loader.js:5322:13)

**Reply by Toilal** [reply]
Le problème disparaît si je désactive l'alias `lodash => lodash-es`

**Reply by FranckFreiburger** [reply]
ok, désactivons `lodash-es` pour le moment

**Reply by FranckFreiburger** [reply]
pb. également avec l'alias `'debug': Path.resolve(__dirname, 'noopDebug.mjs')`
le pb. viens de la mise à jour de babel (mon commit edb406831318182a43075cacba130c1195058651)

je ne comprend pas pourquoi ça déconne :(

**Reply by FranckFreiburger** [reply]
**code généré avec @babel/core@7.13.8 (ok)**

```
  function _debug() {
       const data = _interopRequireDefault(__webpack_require__(146));
       _debug = function() {
           return data;
       }
       ;
       return data;
   }

   ...

   function _interopRequireDefault(obj) {
       return obj && obj.__esModule ? obj : {
           default: obj
       };
   }
   const debug = (0,
   _debug().default)("babel:config:config-chain");
```


**code généré avec @babel/core@7.13.10 (pas ok)**

```
   const buildDebug = __webpack_require__(146);
   const debug = buildDebug("babel:config:config-chain");
```


Peut-être ce commit babel : https://github.com/babel/babel/pull/12963  ou celui là : https://github.com/babel/babel/pull/12795 ?

**Reply by FranckFreiburger** [reply]
pb. réglé : https://github.com/FranckFreiburger/vue3-sfc-loader/commit/635d09569ed3ebb4a8a7486e9e413c900d19706e

**Comment by FranckFreiburger** [comment]
Après ma modif https://github.com/FranckFreiburger/vue3-sfc-loader/commit/d1a1e9ed175683defe64ddb4ea7a28f4d51513e4, je remplacerai bien le contenu de `vue2StyleProcessors.ts` par :

```
export interface StylePreprocessor {
  render(
    source: string,
    map: any | null,
    options: any
  ): StylePreprocessorResults
}

export interface StylePreprocessorResults {
  code: string
  map?: any
  errors: Array<Error>
}

export const processors: { [key: string]: StylePreprocessor } = Object.create(null);
```

du coup, l'utilisateur est totalement libre. T'en penses quoi ?

**Reply by Toilal** [reply]
Est-ce que ça a vraiment du sens d'apporter un StyleProcessor qui ne fonctionnerait même pas dans une véritable application VueJS ? Pour ma part, je préférai ne pas dépasser ce que supporte officiellement VueJS dans Vite, Vue CLI ou vue-loader.

**Reply by FranckFreiburger** [reply]
Très juste
